### PR TITLE
Add support for a helia container

### PR DESCRIPTION
**Pull request by dennis-tra**
Relates to: https://github.com/plprobelab/network-measurements/issues/58

### CODE REVIEW

File: cmd_run.go [hunk 1]
```diff
@@ -79,21 +85,21 @@ var RunCommand = &cli.Command{
 			EnvVars: []string{"TIROS_RUN_DATABASE_SSL_MODE"},
 		},
 		&cli.StringFlag{
-			Name:    "kubo-host",
-			Usage:   "port to reach the Kubo Gateway",
-			EnvVars: []string{"TIROS_RUN_KUBO_HOST"},
+			Name:    "ipfs-host",
```
Comment (SgtPooki) [inline cmd_run.go:88]:
```suggestion
			Name:    "helia-host",
```

Comment (SgtPooki) [reply cmd_run.go:88]: not sure if this is per instance or the global name

Comment (dennis-tra) [reply cmd_run.go:88]: This is per instance and would point to either Kubo or Helia. That's why I renamed it. However, I can't find any usage of that config flag anyway :D

**Comment by SgtPooki** [comment]
@dennis-tra are you waiting on anything from @ipfs/helia-dev to get this merged? Is there anything we can do to help this along?

File: docker-compose.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@ version: "3.9"
 name: tiros
 services:
   ipfs:
-    image: ipfs/kubo:v0.19.0
```
[the remark this answers is not in the document]
Comment (dennis-tra) [reply docker-compose.yml:5]: The idea was to indeed run the latest version of kubo and the most common one based on our measurements. In the past, this has been 0.18 and 0.19. While you're correct that we weren't running the very latest version of Kubo, his docker compose file doesn't represent our deployment. The current deployment configuration is [here](https://github.com/plprobelab/probelab-infra/blob/c2d681bd255082322203e4d259717724fc6df2b5/aws/tf/tiros.tf#L3) and with the new helia version the config looks like [this](https://github.com/plprobelab/probelab-infra/pull/75/files#diff-c06b72be86408466fc04caade3b18c510ebc0359aaaeb76c3ce9f5a4730df647)

[the remark this answers is not in the document]
Comment (BigLep) [reply docker-compose.yml:5]: Thanks for the info.  I can't click through to the link.  Can you please give me access?

**Comment by dennis-tra** [comment]
@SgtPooki the helia image didn't fully work for me yet. That's why I haven't deployed it. I'm in touch with Nishant about it and as soon as the websites and all resources are loaded properly I'll deploy it 👍

File: docker-compose.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (dennis-tra) [reply docker-compose.yml:5]: Done 👍

**Comment by dennis-tra** [comment]
Removed all Kubo references and made it generic on the IPFS implementation. The new database migrations also allow to track which IPFS implementation was under test as there was the hidden (or not so hidden) assumption that it's Kubo.

**Comment by dennis-tra** [comment]
As discussed with @whizzzkid I'll deploy the current version of the Helia image while he works on the remaining bits. This allows us gain early experience in running Helia alongside Kubo in our measurement infra.

File: cmd_run.go [hunk 3]
```diff
@@ -242,12 +262,13 @@ func RunAction(c *cli.Context) error {
...
-	version, sha, err := t.kubo.Version()
+	version, sha, err := t.ipfs.Version()
 	if err != nil {
-		return nil, fmt.Errorf("kubo api offline: %w", err)
+		return nil, fmt.Errorf("ipfs api offline: %w", err)
 	}
 
-	dbRun, err := t.dbClient.InsertRun(c, fmt.Sprintf("%s-%s", version, sha))
+	ipfsImpl := c.String("ipfs-implementation")
+	dbRun, err := t.dbClient.InsertRun(c, ipfsImpl, fmt.Sprintf("%s-%s", version, sha))
```
Comment (SgtPooki) [inline cmd_run.go:271]: FYI @dennis-tra this is failing for me locally saying `ERRO[0000] error: init run: ipfs api offline: version: command not found`. Is this trying to call 'ipfs' binary locally to return version?

Comment (dennis-tra) [reply cmd_run.go:271]: It's not using the local Kubo binary. It's using the HTTP API exposed by Kubo. In this case, the must be an endpoint that implements this specification:  https://docs.ipfs.tech/reference/kubo/rpc/#api-v0-version

**Pull request merged**